content/chat-quick-start/js.md

```diff
@@ -0,0 +1,240 @@
+? Please choose your application name <your-app-name>
+? Which style language do you want to use? css/sass/scss/less/stylus
+? Enable postcss? yes/no
+
+# launch the template app
+npm start
+```
+Now, open the browser and navigate to http://localhost:8000/. If the template project is set up successfully, you should see the following screen.
+
+![Skygear](https://docs.skygear.io/assets/js/js-app-preview.png)
+
+
+## What's in the template project
+
+Before building the chat app, let's first understand the elements in the the template project. They are the very key elements we need for every Skygear app.
+
+#### 1. Configuration
+
+For your client app to communicate with the Skygear server, you need to configure the API key and the app end point in the client.
+
+In the template project, we have put the configuration in ```src/index.js``` and your app information in ```config/base.js```.
+
+Codes in ```src/index.js#L9-15```:
+
+```javascript
+//Configure your Skygear app
+skygear.config({
+  apiKey: config.skygear.apiKey,
+  endPoint: config.skygear.endPoint
+}).then(() => {
+  // Render the main component into the dom
+  ReactDOM.render(<App />, document.getElementById('app'));
+});
+```
+
+Codes in ```config/base.js#L4-9```:
+
+```javascript
+//Input your app info here. You can get them from the Skygear portal
+export default {
+  'skygear': {
+    'endPoint': 'https://<your_app_name>.skygeario.com/',
+    'apiKey': '<your_app_api_key>'
+  }
+}
+```
+
+#### 2. User authentication
+
+A normal chat app requires a user authentication system.
+
+In ```components/Main.js``` in the template project, you can find all the authentication related functions.
+
+Codes in ```components/Main.js#L18-46```:
+
+```javascript
+//user login
+doLogin() {
+  let username = this.refs.username.value;
+  let password = this.refs.password.value;
+
+  skygear.loginWithUsername(username, password).then(() => {
+    this.setState({
+      'user': 'Logged in as '+username
+    });
+  }, () => {
+    this.setState({
+      'user': 'Login fails'
+    });
+  });
+ }
+//user signup
+doSignup() {
+  let username = this.refs.username.value;
+  let password = this.refs.password.value;
+
+  skygear.signupWithUsername(username, password).then(() => {
+    this.setState({
+      'user': 'Signup as '+username
+    });
+  }, () => {
+    this.setState({
+      'user': 'Signup fails'
+    });
+  });
+}
+```
+
+Tips: Even if your app does not require a user system, as every [record](https://docs.skygear.io/guides/cloud-db/basics/js/) in the Skygear database must be owned by a user, you need to at least create a user to perform CRUD. A user can be anonymous. More info [here](https://docs.skygear.io/guides/auth/basics/js/).
+
+## Using Skygear chat in your app
+To call the Skygear chat APIs in your app, first you need to enable the chat function in the _plug-ins_ page in the [developer portal](https://portal.skygear.io/).
+
+![skygear](http://i.imgur.com/FnGjFqv.png)
```

Comment:
we have our own image folder in the repo, you can put the image there instead of using an external link